packages/office-ui-fabric-react/src/components/ResizeGroup/examples/ResizeGroup.OverflowSet.Example.tsx

```diff
@@ -41,152 +48,104 @@ function generateData(count: number, cachingEnabled: boolean, checked: boolean):
-          onReduceData={this._onReduceData}
-          onGrowData={onGrowDataEnabled ? this._onGrowData : undefined}
-          // tslint:disable-next-line:jsx-no-lambda
-          onRenderData={data => {
-            return (
-              <OverflowSet
-                role="menubar"
-                items={data.primary}
-                overflowItems={data.overflow.length ? data.overflow : null}
-                onRenderItem={item => {
-                  return (
-                    <CommandBarButton
-                      role="menuitem"
-                      text={item.name}
-                      iconProps={{ iconName: item.icon }}
-                      onClick={item.onClick}
-                      checked={item.checked}
-                    />
-                  );
-                }}
-                onRenderOverflowButton={overflowItems => {
-                  return <CommandBarButton role="menuitem" menuProps={{ items: overflowItems! }} />;
-                }}
-                styles={overflowSetStyles}
-              />
-            );
-          }}
-        />
-        <div className={styles.settingsGroup}>
-          <Checkbox label="Enable caching" onChange={this._onCachingEnabledChanged} checked={cachingEnabled} />
-          <Checkbox label="Set onGrowData" onChange={this._onGrowDataEnabledChanged} checked={onGrowDataEnabled} />
-          <Checkbox label="Buttons checked" onChange={this._onButtonsCheckedChanged} checked={buttonsChecked} />
-          <div className={styles.itemCountDropdown}>
-            <Dropdown
-              label="Number of items to render"
-              selectedKey={numberOfItems.toString()}
-              onChange={this._onNumberOfItemsChanged}
-              options={[
-                { key: '20', text: '20' },
-                { key: '30', text: '30' },
-                { key: '40', text: '40' },
-                { key: '50', text: '50' },
-                { key: '75', text: '75' },
-                { key: '100', text: '100' },
-                { key: '200', text: '200' },
-              ]}
-            />
-          </div>
-        </div>
-      </div>
-    );
-  }
-
-  private _onReduceData = (currentData: any): any => {
+};
+const short = false;
+const onRenderItem = (item: any) => (
+  <CommandBarButton
+    role="menuitem"
+    text={item.name}
+    iconProps={{ iconName: item.icon }}
+    onClick={item.onClick}
+    checked={item.checked}
+  />
+);
+
+export const ResizeGroupOverflowSetExample: React.FunctionComponent = () => {
+  const [numberOfItems, setNumberOfItems] = React.useState(20);
+  const [buttonsChecked, { toggle: toggleButtonsChecked }] = useBoolean(false);
+  const [cachingEnabled, { toggle: toggleCachingEnabled }] = useBoolean(false);
+  const [onGrowDataEnabled, { toggle: toggleOnGrowDataEnabled }] = useBoolean(false);
+  const dataToRender = generateData(numberOfItems, cachingEnabled, buttonsChecked);
+
+  const onReduceData = (currentData: any): any => {
```

Comment:
Add a type here?

Reply:
```tsx
private _onReduceData = (currentData: any): any => {
```

Not arguing against the suggestion, it's a good one, but the whole ResizeGroup typings are full of anys, and the code here is just a port of what was there.

```tsx

  /**
   * Initial data to be passed to the `onRenderData` function. When there is no `onGrowData` provided, this data should
   * represent what should be passed to the render function when the parent container of the ResizeGroup is at its
   * maximum supported width. A `cacheKey` property may optionally be included as part of the data. Two data objects
   * with the same `cacheKey` will be assumed to take up the same width and will prevent measurements.
   * The type of `cacheKey` is a string.
   */
  data: any;

  /**
   * Function to render the data. Called when rendering the contents to the screen and when
   * rendering in a hidden div to measure the size of the contents.
   */
  onRenderData: (data: any) => JSX.Element;

  /**
   * Function to be performed on the data in order to reduce its width and make it fit into the given space.
   * If there are no more scaling steps to apply, it should return undefined to prevent
   * an infinite render loop.
   */
  onReduceData: (prevData: any) => any;
```